mantidproject/mantid
# Update PDFFourierTransform to feature reverse calculation.

This PR is regarding the implementation of a reverse transform in PDFFourierTransform. This was implemented to make for a smoother fourier transform in the Polaris script, the implementation also included in this PR

**To test:**
First test that the polaris script fourier filter works, this script below requires config files I can supply. Compare the pdf output from this script with and without `freq_params`, the pdf should go to zero at the value given as the paramiter.

Next test transforming forward and backward on the merged and pdf workspaces, detail is lost on double transformations is the bin width is too large or the maximum is too small.

```
from mantid.simpleapi import *

import numpy as np
import matplotlib.pyplot as plt
from isis_powder import polaris, SampleDetails

config_file_path = r"C:/Users/wey38795/Documents/polaris-calculate-pdf/polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="Test", mode="PDF")  

sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si')
polaris.set_sample_details(sample=sample_details)

polaris.create_vanadium(first_cycle_run_no="98532", multiple_scattering=False)
polaris.focus(run_number="98533", input_mode='Summed')

q_lim = [[2.5, 3, 4, 6, 7], [3.5, 5, 7, 11, 40]]
polaris.create_total_scattering_pdf(run_number='98533', 
                                    merge_banks=True, 
                                    q_lims=q_lim, 
                                    freq_params=[1.0])
```
Fixes #28596 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
